agent/internal/client/mock.go

```diff
@@ -528,5 +534,8 @@ func (c *Mock) GetPullRequestInfo(ctx context.Context, taskData TaskData, prNum
 }
 
 func (c *Mock) CreateInstallationToken(ctx context.Context, td TaskData, owner, repo string) (string, error) {
-	return "token", nil
+	if c.CreateInstallationTokenFail {
+		return "", nil
+	}
+	return MockedGitHubAppToken, nil
```

Comment:
For consistency with how the mock is usually implemented, a more common/testing-friendly mock would be to have a field `c.CreateInstallationToken` that the test can set that as the returned token. Like this:
```go
// In agent/command/git_test.go
func (s *GitGetProjectSuite) SetupTest() {
    s.comm.CreateInstallationTokenResult = "mocked_github_app_token"
}

// Tests then will always get that token back (unless you set CreateInstallationTokenFail)

// In the mock
func (c *Mock) CreateInstallationToken(...) {
    if c.CreateInstallationTokenFail { /* return error */ }
    return c.CreateInstallationTokenResult, nil
}
```

It would also let you explicitly encode the various cases of what `CreateInstallationToken` can return like returning an error, vs. returning an empty result, vs. returning a non-empty token.

Reply:
Makes sense! Keeping the mock simple and just returning the values that the test sets is preferable compared to adding a lot of special business logic like validating inputs.